### DISCUSSION TITLE
Remove unnecessary null pointer checks [8617]

### DISCUSSION THREAD

**Discussion by elfring** [eProsima/Fast-DDS#3458] [Ideas]
[An extra null pointer check is not needed in functions](https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null "Do I need to check for null before delete p?") like the following.
- [ReaderProxy](https://github.com/eProsima/Fast-RTPS/blob/7cf8d8d34c114513a913af81eaeb5f966ce53b80/src/cpp/rtps/writer/ReaderProxy.cpp#L69 "Destructor for ReaderProxy")
- [RTPSAsSocketReader](https://github.com/eProsima/Fast-RTPS/blob/7cf8d8d34c114513a913af81eaeb5f966ce53b80/test/blackbox/RTPSAsSocketReader.hpp#L108 "Destructor for RTPSAsSocketReader")
- [WLP](https://github.com/eProsima/Fast-RTPS/blob/80c6ae1ded25d7843f4986fe58eb0287af176239/src/cpp/rtps/builtin/liveliness/WLP.cpp#L90 "Destructor for WLP")

**Comment by JLBuenoLopez** [comment]
While formally correct, the check is also not preventing the expected behavior of the library. Consequently, it is not going to be fixed for the moment. Labeling it as an `enhancement` and moving to the Ideas forum discussion as per [Fast DDS Contributing guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support)

**Reply by elfring** [reply]
:thought_balloon: Would you become interested to [use a development tool like “clang-tidy” for corresponding source code adjustments](https://clang.llvm.org/extra/clang-tidy/checks/readability/delete-null-pointer.html "Deletion of unnecessary null pointer checks")?
